### PR TITLE
Fix issue #52

### PR DESCRIPTION
**Pull request by jamespic**
Fix for issue #52 - uses regex based on ObjectName spec.

### CODE REVIEW

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 1]
```diff
@@ -3,15 +3,12 @@
...
 
 import java.lang.management.ManagementFactory;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: All dependencies should be explicitly listed so you can figure out where they're coming from.

Comment (jamespic) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: No problem. My IDE has a tendency to collect together dependencies, but I'll uncollect them.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 2]
```diff
@@ -137,17 +151,19 @@ private void scrapeBean(MBeanServerConnection beanConn, ObjectName mbeanName) {
...
-        if (idx > 0) {
-            String[] tokens = beanName.substring(idx).split(",|=");
-            for (int i=0; i<tokens.length-1; )
-                output.put(tokens[i++], tokens[i++]);
+        String properties = mbeanName.getKeyPropertyListString();
+        Matcher match = PROPERTY_PATTERN.matcher(properties);
+        while (match.lookingAt()) {
+            output.put(match.group(1), match.group(2));
+            properties = properties.substring(match.end());
+            if (properties.startsWith(","))
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: You should use braces here for clarity

**Comment by brian-brazil** [comment]
That generally looks good, thanks.

**Comment by jamespic** [comment]
I've squashed in the changes you asked for

**Comment by brian-brazil** [comment]
Great, thanks!

**Pull request merged**